src/edit/main.js

```diff
@@ -156,11 +156,22 @@ class ProseMirror {
       domDrop: new DOMSubscription
     }
 
-    if (opts.place && opts.place.appendChild)
+    if (opts.place && opts.place.appendChild) {
+      this.root = opts.place
+
+      // loop if root is not document (in light dom)
+      // only shadowRoot has property (in shadow dom)
+      while (this.root!== document && !this.root.host) {
```

Comment:
`opts.place` doesn't change `this.wrapper`. if `opts.place` is a DOM Element, `this.root` will be the `shadowRoot`. If it's a function then let the function handles `this.wrapper`.